src/ts/src/data/samples/index.tsx

```diff
@@ -1,95 +1,93 @@
+}
 
-const dummySamples: Array<Sample> = [
-    {
-        privateId: "0865-0004KGK00-001",
-        publicId: "0909EEEE-55-33",
-        uploadDate: "2/1/2021",
-        collectionDate: "12/12/2020",
-        collectionLocation: "Santa Clara County",
-        gisaid: "Accepted",
-    },
-    {
-        privateId: "0865-0004KGK00-002",
-        publicId: "0909EEEE-55-34",
-        uploadDate: "2/1/2021",
-        collectionDate: "12/12/2020",
-        collectionLocation: "Santa Clara County",
-        gisaid: "Accepted",
-    },
-];
+// const dummySamples: Array<Sample> = [
+//     {
+//         privateId: "0865-0004KGK00-001",
+//         publicId: "0909EEEE-55-33",
+//         uploadDate: "2/1/2021",
+//         collectionDate: "12/12/2020",
+//         collectionLocation: "Santa Clara County",
+//         gisaid: "Accepted",
+//     },
+//     {
+//         privateId: "0865-0004KGK00-002",
+//         publicId: "0909EEEE-55-34",
+//         uploadDate: "2/1/2021",
+//         collectionDate: "12/12/2020",
+//         collectionLocation: "Santa Clara County",
+//         gisaid: "Accepted",
+//     },
+// ];
 
-const Samples: FunctionComponent<Props> = ({ data = dummySamples }: Props) => {
-    const headerRow = TABLE_HEADERS.map((column) => (
-        <Table.HeaderCell key={column.key}>
-            <div className={style.headerCell}>{column.text}</div>
-        </Table.HeaderCell>
-    ));
+const Samples: FunctionComponent<Props> = ({ data = [] }: Props) => {
+    // we are modifying state using hooks, so we need a reducer
+    const [state, dispatch] = useReducer(searchReducer, {
+        searching: false,
+        results: data,
+    });
 
-    const sampleRow = (sample: Sample): Array<JSX.Element> => {
-        return TABLE_HEADERS.map((column, index) => {
-            let displayData = sample[column.key];
-            let icon: JSX.Element | null = null;
-            if (displayData === undefined) {
-                displayData = UNDEFINED_TEXT;
-            }
-            if (index === 0) {
-                icon = <SampleIcon className={style.icon} />;
-            }
-            return (
-                <Table.Cell key={`${sample.privateId}-${column.key}`}>
-                    <div className={style.cell}>
-                        {icon}
-                        {displayData}
-                    </div>
-                </Table.Cell>
+    // search functions
+    const searcher = (
+        event: React.ChangeEvent<HTMLInputElement>,
+        fieldInput: InputOnChangeData
+    ) => {
+        const query = fieldInput.value;
+        if (query.length === 0) {
+            dispatch({ results: data });
+            return;
+        }
+
+        dispatch({ searching: true });
+
+        const regex = new RegExp(escapeRegExp(query), "i");
+        const filteredSamples = data.filter((sample) => {
+            let result = false;
+            Object.values(sample).forEach(
+                (value) => (result = result || regex.test(value))
```

Comment:
can you short circuit and exit the loop once something tests positive?

Reply:
Yes, this should be a `some()`